### DISCUSSION TITLE
Max retry database connexion

### DISCUSSION THREAD

**Discussion by innovortex** [tiredofit/docker-db-backup#351] [Q&A]
Hi,
Is it possible to limit the number of connection attempts to a database in case of failure ?
Currently, if the connection fails, the program retries indefinitely and does not produce an error message (post-script):

`
 "mysql" 

                counter=0
                export MYSQL_PWD=${backup_job_db_pass}
                while ! (run_as_user mysqladmin -u"${backup_job_db_user}" -P"${backup_job_db_port}" -h"${backup_job_db_host}" ${mysql_tls_args} status > /dev/null 2>&1) ; do
                    sleep 5
                    (( counter+=5 ))
                    write_log warn "MySQL/MariaDB Server '${backup_job_db_host}' is not accessible, retrying.. (${counter} seconds so far)"
                done
            ;;
`

**Comment by innovortex** [comment]
And the problem is that if one of the databases is down, the subsequent ones are not processed. With only one backup per day, this is an issue...

**Reply by tiredofit** [reply]
Have a peek at the (DBXX|DEFAULT)_SKIP_AVAILABILITY_CHECK to bypass that check...

**Comment by innovortex** [comment]
Thank you for this information, but unless I really don't understand, it's impossible to report the error via Discord, for example. Right?

ERROR 2002 (HY000): Can't connect to server on 'x.x.x.x' (115)
2024-06-05.12:52:02 [INFO] ** [05-x.x.x.x__ALL] Backup 05 routines finish time: 2024-06-05 12:52:02 CEST with **exit code 0**
2024-06-05.12:52:02 [NOTICE] ** [05-x.x.x.x__ALL] Backup 05 routines time taken: Hours: 0 Minutes: 00 Seconds: 00


For now, I'm using this modification...

`
if ! (run_as_user mysqladmin -u"${backup_job_db_user}" -P"${backup_job_db_port}" -h"${backup_job_db_host}" ${mysql_tls_args} status > /dev/null 2>&1); then
                    /assets/custom-scripts/discord.sh error $backup_job_db_nom
                    write_log warn "MySQL/MariaDB Server '$backup_job_db_host' is not accessible, retrying.. ($counter seconds et $msg_erreur seconds)"
                fi
`
